src/main/java/org/plp/implementations/plpisa/PlpFile.java

```diff
@@ -0,0 +1,132 @@
+package org.plp.implementations.plpisa;
+
+import org.plp.isa.AsmFile;
+import org.plp.isa.AsmProgram;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This is an in memory representation of an asm file {@link AsmFile}.
+ * This will map to a file path and will be part of {@link PlpProgram}
+ */
+public class PlpFile implements AsmFile {
+    private final String fileName;
+    private List<String> fileContent;
+    private final AsmProgram program;
+
+    public PlpFile(String fileName, AsmProgram program) throws IOException {
+        validateFileName(fileName, program);
+        this.fileName = fileName;
```

Comment:
this.filePath = program.getProgramDirectory() + fileName;

This constructor is tricky if it only invokes inside the AsmProgram object ("createAsmFileInProgram"). We don't need to know the program at all. The parameter to the constructor should be a full path.

And the name validation will be performed in "AsmProgram::createAsmFileInProgram".

AsmProgram class is managing all the files.

Reply:
Yes going to make AsmFile independent of the AsmProgram. Program will be managing files but files itself do not need to know about the program.